### DISCUSSION TITLE
Problema Actividad 2 - Grupo 2

### DISCUSSION THREAD

**Discussion by ValentinMilocco** [frlp-utn-ingsoft/catedra#123] [Q&A] [answered]
Tengo un problema a la hora de realizar la tarea:  Validar que la fecha de nacimiento de la mascota sea menor al día de hoy.
pude hacer el test de unidad por lo cual la validacion es correcta, pero no pude modificar el view de forma tal para que aparezca el error por interfaz.

**Comment by RodrigoJacznik** [comment]
Buenas. Que error estas teniendo?

**Reply by ValentinMilocco** [reply]
No estoy pudiendo mostrar el error por pantalla, del mismo estilo que aparece, "Por favor ingrese un nombre" cuando la casilla de nombre esta vacia.
El error lo guarda porque puedo imprimirlo por consola, pero no puedo hacer que se muestre en la interfaz.

**Reply by RodrigoJacznik** [reply]
Que código hiciste hasta ahora? Necesito un poco de código para poder ayudarte.

**Reply by ValentinMilocco** [reply]
Modifique el archivo "models" tanto la funcion validate_pet, para pasarle el error, a su vez modifique la funcion update_pet de la siguiente forma:    def update_pet(self, pet_data):
        self.name = pet_data.get("name", "") or self.name
        self.breed = pet_data.get("breed", "") or self.breed
        new_birthday = pet_data.get("birthday", "") or self.birthday
        if new_birthday:
            birthday_date = datetime.strptime(new_birthday, "%Y-%m-%d").date()
            if birthday_date > date.today():
                raise ValueError("La fecha de nacimiento no puede ser posterior al día actual.")
        self.birthday = new_birthday
        
        self.save()
        Y pasa el test de unidad, pero aun asi cuando quiero editar o agregar una mascota con la fecha posterior al dia actual simplemente mi aparece un tilde verde pero no me deja seguir y ahi es donde deberia aparecer la cruz roja con el error "la fecha de nacimiento no puede ser posterior a la actual", que se obtiene del validate_pet de la siguiente forma:
            if birthday == "":
        errors["birthday"] = "Por favor ingrese una fecha de nacimiento"
    else:
        try:
            birthday_date = datetime.strptime(birthday, "%Y-%m-%d").date()
            if birthday_date > date.today():
                errors["birthday"] = "La fecha de nacimiento no puede ser posterior al día actual."

**Reply by ValentinMilocco** [reply]
perdon no me di cuenta de pasarte el link de la rama:
https://github.com/JuaniLezcano/vetsoft/tree/Feature/Pet-birthday

**Reply by RodrigoJacznik** [reply, marked as answer]
El código lo veo bien, falta el atributo max en el template para que bootstrap tome mejor las validaciones. Te debería quedar así:

![image](https://github.com/frlp-utn-ingsoft/catedra/assets/2715743/87d1f80e-aed9-47e9-ad54-2cfbf02d15da)
